internal/state/dataplane/configuration.go

```diff
@@ -83,13 +83,43 @@ type MatchRule struct {
 	// the entire resource.
 	Source *v1beta1.HTTPRoute
 	// BackendGroup is the group of Backends that the rule routes to.
-	BackendGroup graph.BackendGroup
+	BackendGroup BackendGroup
 	// MatchIdx is the index of the rule in the Rule.Matches.
 	MatchIdx int
 	// RuleIdx is the index of the corresponding rule in the HTTPRoute.
 	RuleIdx int
 }
 
+// BackendGroup represents a group of Backends for a routing rule in an HTTPRoute.
```

Comment:
Sure, I can remove the comments about route and keep the source generic.

Reply:
could we also update the comment for `func (bg *BackendGroup) Name() string ` ?

Reply:
The name will only be unique across all HTTPRoutes. If we add support for another route type, that function must change to prevent collisions. We would need to add a field to the BackendGroup that held the type of route to prevent collisions. I hesitate to make this change when we only support HTTPRoutes right now. 

I think we should keep the HTTPRoute comments for both the BackendGroup and Name() method until we introduce a new route type. Once we get to that point, we will better understand how to make BackendGroup more extensible. These comments will give us a heads-up that changes are needed for other route types.